### DISCUSSION TITLE
Installation failure (compat-30)

### DISCUSSION THREAD

**Discussion by harold** [magit/magit#5246] [Support] [answered]
Hi, coming back to magit after many years - thank you for your tireless efforts, I am excited to get back into the fold. (:

`emacs-version`:

```
GNU Emacs 27.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.33, cairo version 1.16.0) of 2024-09-17, modified by Debian1
```

I tried to install magit today from `melpa-stable` today with `package-refresh-contents` and `package-install` `magit`.

fwiw, I have:

```
(add-to-list 'package-archives '("melpa-stable" . "http://stable.melpa.org/packages/") t)
```

I got this error:

```
package-compute-transaction: Need package ‘compat-30.0.0.0’, but only 29.1.4.4 is available
```

Now `package-list-packages` shows this:

```
magit              4.1.1         incompat   melpa-stable A Git porcelain inside Emacs
```

And the details are this:

```
Package magit is incompatible.

     Status: Incompatible because it depends on uninstallable packages.
    Archive: melpa-stable
    Version: 4.1.1
     Commit: 93e86ceca7bf5b0ff9023d7f138e5f06990fc276
    Summary: A Git porcelain inside Emacs
   Requires: emacs-26.1, compat-30.0.0.0 (not available), dash-2.19.1, magit-section-4.1.1, seq-2.24, transient-0.7.6,
               with-editor-3.4.2
   Homepage: https://github.com/magit/magit
   Keywords: git tools vc 
 Maintainer: Jonas Bernoulli <emacs.magit@jonas.bernoulli.dev>
    Authors: Marius Vollmer <marius.vollmer@gmail.com>
             Jonas Bernoulli <emacs.magit@jonas.bernoulli.dev>
```

Thanks in advance for your time and consideration, let me know if there's any other info I can provide.

**Comment by tarsius** [comment]
The `compat` package is from GNU ELPA. Apparently your local list of packages available from that *ELPA is severely out-of-date. Update it with `M-x package-refresh-contents`.

**Reply by harold** [reply]
Ah, I had tried that first, but I didn't notice there was an error there as well:

```
Importing package-keyring.gpg...done
Contacting host: elpa.gnu.org:443 [2 times]
Failed to download ‘gnu’ archive.
```

Apparently, my setup is differently broken! I will investigate. Thank you.

**Reply by harold** [reply, marked as answer]
Thank you for the pointer - getting the gnu archive up to date fixed it.

Sorry for the noise.

:bow: - stay rad.
